src/traversal/mod.rs

```diff
@@ -608,15 +625,21 @@ pub fn graph_dfs_search(
 ///     a default value of cost ``1.0`` will be used for each edge.
 /// :param visitor: A visitor object that is invoked at the event points inside the
 ///     algorithm. This should be a subclass of :class:`~rustworkx.visit.DijkstraVisitor`.
+///     This has a default value of ``None`` as a backwards compatibility artifact (to
+///     preserve argument ordering from an earlier version) but it is a required argument
+///     and will raise a ``TypeError`` if not specified.
 #[pyfunction]
-#[pyo3(text_signature = "(graph, source, weight_fn, visitor)")]
 pub fn digraph_dijkstra_search(
     py: Python,
     graph: &digraph::PyDiGraph,
     source: Option<Vec<usize>>,
     weight_fn: Option<PyObject>,
-    visitor: PyDijkstraVisitor,
+    visitor: Option<PyDijkstraVisitor>,
 ) -> PyResult<()> {
+    if visitor.is_none() {
+        return Err(PyTypeError::new_err("Missing required argument visitor"));
+    }
+    let v = visitor.unwrap();
```

Comment:
```suggestion
    let visitor = visitor.unwrap();
```
I'd rather do this than change the variable name